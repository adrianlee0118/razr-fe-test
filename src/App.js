import React, { useState, useEffect } from "react";
import Circle from "./components/circle";
import Square from "./components/square";

import useGenerateFunction from "./hooks/useGenerateFunction";
import { Random_100_URL } from "./constants";

//App component -- handles calls to rendering of shapes determined by JSON object list passed from useGenerateFunction React hook, manages dynamic rotation of squares
const App = () => {
  const { shapes } = useGenerateFunction(Random_100_URL);
  const [angle, setAngle] = useState(0);

  //Rotate squares clockwise by 10 degrees every 500 ms ([angle] indicates a re-render is to be triggered whenever angle changes)
  useEffect(() => {
    const interval = setInterval(
      () => setAngle(angle === 80 ? 0 : angle + 10),
      500
    );
    return () => clearInterval(interval);
  }, [angle]);

  return (
    <div
      className="App"
      style={{
        "text-align": "center",
      }}
    >
      <h1>Feast your Eyes</h1>
      <div>
        {shapes.length === 0 ? (
          <p>No data: something went wrong.</p>
        ) : (
          shapes.map((shape) =>
            shape.type === "circle" ? (
              <Circle key={shape.key} radius={shape.crossdim / 2} />
            ) : (
              <Square key={shape.key} sidelen={shape.crossdim} angle={angle} />
            )
          )
        )}
      </div>
    </div>
  );
};

export default App;
