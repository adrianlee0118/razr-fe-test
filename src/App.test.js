import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  //Component test
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  test("has a valid snapshot", () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders one container for all shape items", () => {
    const element = shallow(<App />); //shallow render, no child components
    expect(element.find(".App").length).toBe(1);
  });
});
