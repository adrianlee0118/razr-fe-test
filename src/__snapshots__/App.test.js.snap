// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="App"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Feast your Eyes
  </h1>
  <div>
    <p>
      No data: something went wrong.
    </p>
  </div>
</div>
`;
